lib/Models/FunctionParameters/NumberParameter.ts

```diff
@@ -0,0 +1,72 @@
+import { computed, makeObservable, override } from "mobx";
+import FunctionParameter, {
+  Options as FunctionParameterOptions
+} from "./FunctionParameter";
+import CatalogFunctionMixin from "../../ModelMixins/CatalogFunctionMixin";
+
+interface Options extends FunctionParameterOptions {
+  minimum?: number;
+  maximum?: number;
+  defaultValue?: number;
+}
+
+export default class NumberParameter
+  extends FunctionParameter<string>
+  implements Options
+{
+  static readonly type = "number";
+  readonly type = "number";
+  minimum?: number;
+  maximum?: number;
+  defaultValue?: number;
+
+  @computed
+  get rangeDescription() {
+    if (this.minimum !== undefined && this.maximum !== undefined) {
+      return `: must be between ${this.minimum} to ${this.maximum}`;
+    } else if (this.minimum !== undefined) {
+      return `: at least ${this.minimum}`;
+    } else if (this.maximum !== undefined) {
+      return `: at most ${this.maximum}`;
+    } else {
+      return "";
+    }
+  }
+
+  @override
+  get isValid(): boolean {
+    let value =
+      this.value !== undefined
+        ? parseFloat(this.value)
+        : (this.defaultValue as number);
+
+    if (typeof value === "string") {
+      value = parseFloat(value);
+    }
+
+    if (value === undefined || isNaN(value)) {
+      return !this.isRequired;
+    }
```

Comment:
Just a thought. Here we have to duplicate the checks from parent [isValid implementation](https://github.com/TerriaJS/terriajs/blob/92c0afe822e323804b38beff5479100936d7e532/lib/Models/FunctionParameters/FunctionParameter.ts#L42-L44).

Instead, would it be better if we override the [value getter](https://github.com/TerriaJS/terriajs/blob/92c0afe822e323804b38beff5479100936d7e532/lib/Models/FunctionParameters/FunctionParameter.ts#L50-L52) and in this method we do something equivalent to `super.isValid === false ? false : this.isValidRange()`.

Idea is that, if the parent `isValid` is modified with more validations then we should apply here too.

Reply:
added an isValidRange check and call it in the overriden isValid method